react/week2/todoapp/src/components/TodoList.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import todosData from './todosData';
+import TodoItem from './TodoItem';
+import Timer from './Timer';
+
+class TodoList extends React.Component {
+   constructor(props) {
+      super(props);
+      this.state = {
+         todos: todosData
+      };
+      this.handleChange = this.handleChange.bind(this);
+   }
+   addTodo = () => {
+      const newItem = { description: 'newText' };
+      const newList = this.state.todos.concat(newItem);
+      this.setState({ todos: newList }, () => {
+         console.log('state has been updated');
+      });
+   };
+
+    removeTodo = (index) =>
+    {
+       const todo = Object.assign([], this.state.todos);
+       todo.splice(index, 1);
+       this.setState({ todos: todo });
+       
+    };
+
+   handleChange(id) {
+      this.setState((prevState) => {
+         const updatedTodos = prevState.todos.map((todo) => {
+            if (todo.id === id) {
+               return {
+                  ...todo,
+                  done: !todo.done
+               };
+            }
+            return todo;
+         });
+         return {
+            todos: updatedTodos
+         };
```

Comment:
It's a good practice to include anything else that might be in the state so you don't override anything by mistake:
```suggestion
         return {
            ...prevState,
            todos: updatedTodos
         };
```